### PR TITLE
subithal : Solved All problems.

### PR DESCRIPTION
**Pull request by subithal**

### CODE REVIEW

**Comment by maknahar** [comment]
Hi @subithal, Thanks for submitting the solutions.

I just ran the server and facing the following issue while visiting `http://localhost:8090/problem1` and `http://localhost:8090/problem2`.

Screenshot 2019-10-04 at 12 05 30 PM


Server start just fine.

File: api/service/contentManagerService.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+const fs = require("fs");
+
+function createOutputFile(data,fileName){
+    fastcsv.writeToPath('./assets/'+fileName,data)
+      .on('error',(err)=>console.log(err))
+      .on('finish',()=>console.log('Sucess'))
+  }
+
+  function getPartnersDetails() {
+    var partnerDetails = [];
```
Comment (maknahar) [inline api/service/contentManagerService.js]: Is there any alternative to using `var`?

**Comment by maknahar** [comment]
Changed the partners.json to this file and run the solution for problem 1. The expected and actual does not match could you check?

**Comment by subithal** [comment]
> Changed the partners.json to this file and run the solution for problem 1. The expected and actual does not match could you check?

Sure. 
Could you please share the partners.csv that you are using.
Make sure File format is .csv not .json

**Comment by maknahar** [comment]
Sorry! Missed that. This is the content in partners.csv
```
Theatre,Size Slab (in GB),Minimum cost,Cost Per GB,Partner ID
T1     ,100-200          ,2000        ,13         ,P1
T1     ,0-100            ,1500        ,20         ,P1
T1     ,200-300          ,2500        ,12         ,P1
T1     ,300-400          ,3000        ,10         ,P1
T2     ,0-100            ,1500        ,20         ,P1
T2     ,100-200          ,2000        ,15         ,P1
T2     ,200-300          ,2500        ,12         ,P1
T2     ,300-400          ,3000        ,10         ,P1
T1     ,0-200            ,1000        ,20         ,P2
T1     ,200-400          ,2500        ,15         ,P2
T2     ,0-200            ,2500        ,20         ,P2
T2     ,200-400          ,3500        ,10         ,P2
T1     ,100-200          ,800         ,25         ,P3
T1     ,200-600          ,1200        ,30         ,P3
T2     ,100-200          ,900         ,15         ,P3
T2     ,200-400          ,1000        ,12         ,P3
```

**Comment by subithal** [comment]
> Sorry! Missed that. This is the content in partners.csv
> 
> ```
> Theatre,Size Slab (in GB),Minimum cost,Cost Per GB,Partner ID
> T1     ,100-200          ,2000        ,13         ,P1
> T1     ,0-100            ,1500        ,20         ,P1
> T1     ,200-300          ,2500        ,12         ,P1
> T1     ,300-400          ,3000        ,10         ,P1
> T2     ,0-100            ,1500        ,20         ,P1
> T2     ,100-200          ,2000        ,15         ,P1
> T2     ,200-300          ,2500        ,12         ,P1
> T2     ,300-400          ,3000        ,10         ,P1
> T1     ,0-200            ,1000        ,20         ,P2
> T1     ,200-400          ,2500        ,15         ,P2
> T2     ,0-200            ,2500        ,20         ,P2
> T2     ,200-400          ,3500        ,10         ,P2
> T1     ,100-200          ,800         ,25         ,P3
> T1     ,200-600          ,1200        ,30         ,P3
> T2     ,100-200          ,900         ,15         ,P3
> T2     ,200-400          ,1000        ,12         ,P3
> ```

I sincerely apologize for the delay in response..Could you please just confirm, the attached partner.csv on the problem page and this data are the same.
I'm getting a proper response.
If you are expecting an output in the form of .csv file. the same is available under the assets folder

**Comment by subithal** [comment]
I got that issue..I'll look further into it.

**Comment by subithal** [comment]
> > Sorry! Missed that. This is the content in partners.csv
> > ```
> > Theatre,Size Slab (in GB),Minimum cost,Cost Per GB,Partner ID
> > T1     ,100-200          ,2000        ,13         ,P1
> > T1     ,0-100            ,1500        ,20         ,P1
> > T1     ,200-300          ,2500        ,12         ,P1
> > T1     ,300-400          ,3000        ,10         ,P1
> > T2     ,0-100            ,1500        ,20         ,P1
> > T2     ,100-200          ,2000        ,15         ,P1
> > T2     ,200-300          ,2500        ,12         ,P1
> > T2     ,300-400          ,3000        ,10         ,P1
> > T1     ,0-200            ,1000        ,20         ,P2
> > T1     ,200-400          ,2500        ,15         ,P2
> > T2     ,0-200            ,2500        ,20         ,P2
> > T2     ,200-400          ,3500        ,10         ,P2
> > T1     ,100-200          ,800         ,25         ,P3
> > T1     ,200-600          ,1200        ,30         ,P3
> > T2     ,100-200          ,900         ,15         ,P3
> > T2     ,200-400          ,1000        ,12         ,P3
> > ```
> 
> I sincerely apologize for the delay in response..Could you please just confirm, the attached partner.csv on the problem page and this data are the same.
> I'm getting a proper response.
> If you are expecting an output in the form of .csv file. the same is available under the assets folder

this is the output I'm getting after running my code for the given partners.csv file.

D1,true,P1,2000
D2,true,P1,3250
D3,true,P3,15300
D4,false,,

**Comment by maknahar** [comment]
Hi @subithal,  this is what I am getting as output:

Screenshot 2019-10-09 at 10 20 01 AM
Screenshot 2019-10-09 at 10 20 18 AM

Input Files:

Screenshot 2019-10-09 at 10 23 08 AM

Screenshot 2019-10-09 at 10 22 03 AM

**Comment by maknahar** [comment]
> Could you please just confirm, the attached partner.csv on the problem page and this data are the same.

There is a minor difference. Did you try replacing the existing content with this one?

**Comment by subithal** [comment]
> > Could you please just confirm, the attached partner.csv on the problem page and this data are the same.
> 
> There is a minor difference. Did you try replacing the existing content with this one?

if i'm not wrong... You have changed the D1 content to 250 in the input file?
and you are getting the following output.
D1,true,P1,3000
D2,true,P1,3250
D3,true,P3,15300
D4,false,,

As per observation this is seems to be the right output for the given input.

**Pull request closed without merging**